server/src/main/java/io/crate/expression/scalar/regex/RegexpReplaceFunction.java

```diff
@@ -65,16 +70,16 @@ public static void register(ScalarFunctionModule module) {
     private final Signature signature;
     private final Signature boundSignature;
     @Nullable
-    private final RegexMatcher regexMatcher;
+    private Pattern pattern;
```

Comment:
maybe final or do we really have to mutate it in the `evaluate`?

Reply:
compile creates the Pattern.

Reply:
https://github.com/crate/crate/pull/10104/files#diff-bffb730cd7513f358bd585218cd15286R155

Reply:
evaluate does not mutate pattern, it checks whether it was initialised (compile would do this), and if not, then it initialises it. I kind of preserved this logic from the previous scalar implementation.